src/mumble/MainWindow.cpp

```diff
@@ -504,30 +504,55 @@ bool MainWindow::nativeEvent(const QByteArray &, void *message, long *) {
 #endif
 
 void MainWindow::closeEvent(QCloseEvent *e) {
-#ifndef Q_OS_MAC
 	ServerHandlerPtr sh = Global::get().sh;
-	if (sh && sh->isRunning() && Global::get().s.bAskOnQuit && !bSuppressAskOnQuit) {
+
+#ifndef Q_OS_MAC
+	if (sh && sh->isRunning() && !bSuppressAskOnQuit
+		&& Global::get().s.connectedQuitBehavior == ConnectedQuitBehaviorID::ASK) {
 		QMessageBox mb(QMessageBox::Warning, QLatin1String("Mumble"),
 					   tr("Mumble is currently connected to a server. Do you want to Close or Minimize it?"),
 					   QMessageBox::NoButton, this);
+		QCheckBox   *qcbRemember = new QCheckBox(tr("Remember this setting"));
 		QPushButton *qpbClose    = mb.addButton(tr("Close"), QMessageBox::YesRole);
 		QPushButton *qpbMinimize = mb.addButton(tr("Minimize"), QMessageBox::NoRole);
 		QPushButton *qpbCancel   = mb.addButton(tr("Cancel"), QMessageBox::RejectRole);
 		mb.setDefaultButton(qpbClose);
 		mb.setEscapeButton(qpbCancel);
+		mb.setCheckBox(qcbRemember);
 		mb.exec();
 		if (mb.clickedButton() == qpbMinimize) {
 			showMinimized();
 			e->ignore();
+
+			if (qcbRemember->isChecked()) {
+				Global::get().s.connectedQuitBehavior = ConnectedQuitBehaviorID::MINIMIZE;
+			}
+
 			return;
 		} else if (mb.clickedButton() == qpbCancel) {
 			e->ignore();
 			return;
 		}
+
+		if (qcbRemember->isChecked()) {
+			Global::get().s.connectedQuitBehavior = ConnectedQuitBehaviorID::CLOSE;
+		}
+	} else if (sh && sh->isRunning() && Global::get().s.connectedQuitBehavior == ConnectedQuitBehaviorID::MINIMIZE) {
+		showMinimized();
+		e->ignore();
+		return;
 	}
-	sh.reset();
 #endif
 
+#ifdef Q_OS_MAC
+	if (sh && sh->isRunning() && Global::get().s.connectedQuitBehavior == ConnectedQuitBehaviorID::MINIMIZE) {
+		showMinimized();
+		e->ignore();
+		return;
+	}
+#endif
+	sh.reset();
```

Comment:
This could be condensed like so:
```cpp
} else
#endif //Q_OS_MAC
if (sh && sh->isRunning() && Global::get().s.connectedQuitBehavior == ConnectedQuitBehaviorID::MINIMIZE) {
		showMinimized();
		e->ignore();
		return;
	}
```
That way, we don't have to duplicate the code for the remembered minimization